sdk/cosmosdb/cosmos/src/client/Item/Items.ts

```diff
@@ -484,54 +491,150 @@ export class Items {
       const path = getPathFromLink(this.container.url, ResourceType.item);
 
       const orderedResponses: OperationResponse[] = [];
-      await Promise.all(
-        batches
-          .filter((batch: Batch) => batch.operations.length)
-          .flatMap((batch: Batch) => splitBatchBasedOnBodySize(batch))
-          .map(async (batch: Batch) => {
-            if (batch.operations.length > 100) {
-              throw new Error(
-                "Cannot run bulk request with more than 100 operations per partition",
-              );
-            }
-            try {
-              const response = await addDignosticChild(
-                async (childNode: DiagnosticNodeInternal) =>
-                  this.clientContext.bulk({
-                    body: batch.operations,
-                    partitionKeyRangeId: batch.rangeId,
-                    path,
-                    resourceId: this.container.url,
-                    bulkOptions,
-                    options,
-                    diagnosticNode: childNode,
-                  }),
+      const batchesQueue: Queue<Batch> = new Queue<Batch>();
+      // Enqueue the Batches
+      batches
+        .filter((batch: Batch) => batch.operations.length)
+        .flatMap((batch: Batch) => splitBatchBasedOnBodySize(batch))
+        .map(async (batch: Batch) => {
+          if (batch.operations.length > 100) {
+            throw new Error("Cannot run bulk request with more than 100 operations per partition");
+          }
+          batchesQueue.enqueue(batch);
+        });
+      // Dequeue the Batches and execute them one by one
+      while (!batchesQueue.isEmpty()) {
+        const batch = batchesQueue.dequeue();
```

Comment:
As @v1k1 mentioned, we need parallel requests for each partition. Instead of queue, we could initialize all the requests simultaneously, maintain an array of promises and use [allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) to wait and verify the result of each response. If any split/merge we could fire those requests again and wait for response.

Reply:
addressed